### ISSUE TITLE
Refactor Message to Notification

### ISSUE THREAD

**Issue by StefanMaron** [DavidFeldhoff/al-codeactions#147]
Hi David!

I just realized that I could make far more use of Notifcations and that it often makes sense to use a notification instead of a message.
That way the flow in BC can better for the user, as he does not get interrupted by message so often.

What do you think about the idea to refactor A message into a notification.

This:
``` AL
Message('Some message to show to the user');
```
To this:
``` AL
MyNotification.Message('Some message to show to the user');
MyNotification.Send();
```

When a label already is in place, that should be used obviously.

**Comment by waldo1001** [comment]
Hm, may be not in this case.  An ID will be assigned anyway (in memory).  Only interesting to be recalled (or to make sure the same is not being added multiple times)

**Comment by DavidFeldhoff** [comment]
An ID could also be good if you want to turn off the notification, but then it would become a bit more complex. Probably too much, right? Or would you like to have at least the possibility to have the code afterwards like this?
```al
var
    TheMessageTxt: Label 'Some message to show to the user';

procedure ExistingProc()
var
    MyNotifications: Record "My Notifications";
begin
    [some other existing code]
    if MyNotifications.IsEnabled(GetNotificationId()) then begin
        MyNotification.Id := GetNotificationId();
        MyNotification.Message(TheMessageTxt);
        MyNotification.Send();
    end;
    [some other existing code]
end;

local procedure GetNotificationId(): Guid
var
begin
    exit('5dd5a7df-14cc-49b6-9d16-b34066c06f05');
end;

procedure OnInitializingNotificationWithDefaultState()
var
    MyNotifications: Record "My Notifications";
    DescriptionTxt: Label 'Text inserted by snippet';
begin
    MyNotifications.InsertDefault(GetNotificationId(), TheMessageTxt, DescriptionTxt, true);
end;
```

**Comment by DavidFeldhoff** [comment]
It could be two codeactions like:
1. Convert to Notification
2. Convert to optional Notification

**Comment by StefanMaron** [comment]
I like to have both posibilities, the full example and the very minimalistic one :)

**Comment by DavidFeldhoff** [comment]
But at least the "Convert to optional notification" would only work in codeunits right? Because only there I'm able to create event subscribers..

**Comment by StefanMaron** [comment]
I mean, you could still create it so its there to be copied somewhere else, right?

**Comment by DavidFeldhoff** [comment]
Yes, would be possible, but then "suddenly" after a codeaction your solution isn't compilable anymore right? I can imagine that a few people are then a bit confused. If I'd do it like that, I'd at least put a comment to it or something like that

**Comment by StefanMaron** [comment]
Or you could also only offer the complex scenario if the code already is withing a codeunit :)

**Comment by DavidFeldhoff** [comment]
Short update: haven't started with this yet, as I'm a bit pushing the other two extensions currently (ATDD.TestScriptor and Azure DevOps Simplify). Will come to this at a later time
